README.md

```diff
@@ -1,32 +1,144 @@
+1. [What is this repository?](#what_is_it)
+2. [Quickstart and examples](#quickstart)
+3. [Assembling inference model code folder](#inference_model_folder)
+4. [Custom Model Templates](#custom_model_templates)
+5. [Custom Environment Templates](#custom_environment_templates)
+6. [Custom Model Runner (drum)](#custom_model_runner)
+4. [Contribution & development](#contribution_development)
+
+## What is this repository? <a name="what_is_it"></a>
+The **DataRobot User Models** repository contains information and tools for assembling,
+debugging, testing, and running your training and inference models with DataRobot.
+
+#### Terminology
+This repository address the DataRobot functionality known as `custom models`. The terms `custom model` and `user model` can be used interchangeably, as can `custom model directory` and `code directory`.
+
+## Quickstart <a name="quickstart"></a>
+The following example shows how to use the [**drum**](https://github.com/datarobot/datarobot-user-models/tree/master/custom_model_runner) tool to make predictions on an [sklearn model](model_templates/inference/python3_sklearn)
+1. Clone the repository
+2. Create a virtual environment: `python3 -m virtualenv <dirname for virtual environment>`
+3. Activate the virtual environment: `source <dirname for virtual environment>/bin/activate`
+4. cd into the repo: `cd datarobot-user-models`
+5. Install the required dependencies: `pip install -r public_dropin_environments/python3_sklearn/requirements.txt`
+6. Install datarobot-drum: `pip install datarobot-drum`
+7. Run the example: `drum score --code-dir model_templates/inference/python3_sklearn --input tests/testdata/boston_housing.csv`
+
+For more examples, reference the [Custom Model Templates](#custom_model_templates).
+
+## Assembling a custom model folder <a name="inference_model_folder"></a>
+> Note: the following information is only relevant you are using [**drum**](https://github.com/datarobot/datarobot-user-models/tree/master/custom_model_runner) to run a model. 
+
+To create a custom inference model, you must provide specific files to use with a custom environment:
+
+- a serialized model artifact with a file extension corresponding to the chosen environment language.
+- any additional custom code required to use it.
+
+The `drum new model` command can help you generate a model template. Check [here](https://github.com/datarobot/datarobot-user-models/tree/master/custom_model_runner#model-template-generation) for more information.
+
+### Built-In Model Support
+The **drum** tool has built-in support for the following libraries. If your model is based on one of these libraries, **drum** expects your model artifact to have a matching file extension.
+
+### Python libraries
+| Library | File Extension | Example |
+| --- | --- | --- |
+| scikit-learn | *.pkl | sklean-regressor.pkl |
+| xgboost | *.pkl | xgboost-regressor.pkl |
+| PyTorch | *.pth | torch-regressor.pth |
```

Comment:
pytorch and java don't work for training models yet...

Reply:
this is for inference only

Reply:
I'll update section header